qr_generator/lib/screens/scan_view/scan_view.dart

```diff
@@ -0,0 +1,154 @@
+        ],
+      ),
+      body: SafeArea(
+        child: BlocListener<ScanViewBloc, ScanViewStateBloc>(
+            listener: (context, state) {
+          if (state.scanStatus == ScanStatus.loaded) {
+            _showBottomSheet(
+                context,
+                ListTile(
+                  leading: CircleAvatar(
+                      backgroundColor:
+                          state.seedValidated ? Colors.green : Colors.red,
+                      child: Icon(
+                        state.seedValidated ? Icons.check : Icons.close,
+                        color: Colors.white,
+                      )),
+                  title: Text(state.seedValidated ? 'Valid' : 'Invalid'),
+                  subtitle: Text(state.seedData),
+                ));
+          }
+        }, child: BlocBuilder<ScanViewBloc, ScanViewStateBloc>(
+          builder: (context, state) {
+            if (state.scanStatus == ScanStatus.loaded ||
+                state.scanStatus == ScanStatus.initial) {
+              return BlocProvider.value(
+                value: BlocProvider.of<ScanViewBloc>(context),
+                child: buildQrView(context),
+              );
+            }
+            if (state.scanStatus == ScanStatus.failure) {
+              return ErrorView(
+                error: state.errorMessage,
+              );
+            }
+            return const Center(
+              child: CircularProgressIndicator(),
+            );
+          },
+        )),
+      ),
+    );
+  }
+
+  Widget buildQrView(BuildContext context) {
+    var scanArea = (MediaQuery.of(context).size.width < 400 ||
+            MediaQuery.of(context).size.height < 400)
+        ? 150.0
+        : 300.0;
+    // To ensure the Scanner view is properly sizes after rotation
+    // we need to listen for Flutter SizeChanged notification and update controller
+    return QRView(
+      key: qrKey,
+      onQRViewCreated: _onQRViewCreated,
+      overlay: QrScannerOverlayShape(
+          borderColor: Colors.green,
+          borderRadius: 10,
+          borderLength: 30,
+          borderWidth: 10,
+          cutOutSize: scanArea),
+      onPermissionSet: (ctrl, p) => _onPermissionSet(context, ctrl, p),
+    );
+  }
+
+  void _showBottomSheet(BuildContext context, Widget body,
+      {bool isDismissible = true, bool enableDrag = true}) {
+    showModalBottomSheet<void>(
+      isScrollControlled: true,
+      elevation: 3,
+      context: context,
+      isDismissible: isDismissible,
+      enableDrag: enableDrag,
+      clipBehavior: Clip.antiAlias,
+      shape: RoundedRectangleBorder(
+        borderRadius: BorderRadius.circular(10.0),
+      ),
+      builder: (BuildContext context) {
+        return body;
+      },
+    ).then((_) async {
+      await controller.resumeCamera();
+      BlocProvider.of<ScanViewBloc>(context).add(RestartScanData());
+    });
+  }
+
+  void _onQRViewCreated(QRViewController controller) {
+    setState(() {
```

Comment:
**question**: if you're handling the state with bloc, why are you using setState too? There's a mix of state management solutions here which causes inconsistency

Reply:
I understand your reasoning, I'll take it into account in the future.